test/util/expect.ts

```diff
@@ -0,0 +1,145 @@
+  Method extends keyof Event[Section],
+  Tuple extends ExtractTuple<Event[Section][Method]>
+>(
+  block: BlockCreationResponse<ApiType, Calls extends Call[] ? Awaited<Call>[] : Awaited<Call>>,
+  section: Section,
+  method: Method
+): IEvent<Tuple> {
+  let event: EventRecord | undefined;
+  if (Array.isArray(block.result)) {
+    block.result.forEach((r, idx) => {
+      const foundEvents = r.events.filter(
+        ({ event }) => event.section.toString() == section && event.method.toString() == method
+      );
+      if (foundEvents.length > 0) {
+        expect(
+          event,
+          `Event ${section.toString()}.${method.toString()} appeared multiple times`
+        ).toBeUndefined();
+        expect(
+          foundEvents,
+          `Event ${section.toString()}.${method.toString()} appeared multiple times`
+        ).to.be.length(1);
+        event = foundEvents[0];
+      }
+    });
+  } else {
+    const foundEvents = block.result!.events!.filter(
+      ({ event }) => event.section.toString() == section && event.method.toString() == method
+    );
+    if (foundEvents.length > 0) {
+      expect(
+        foundEvents,
+        `Event ${section.toString()}.${method.toString()} appeared multiple times`
+      ).to.be.length(1);
+      event = foundEvents[0];
+    }
+  }
+  expect(
+    event,
+    `Event ${section.toString()}.${method.toString()} not found:\n${(Array.isArray(block.result)
+      ? block.result.map((r) => r.events).flat()
+      : block.result
+      ? block.result.events
+      : []
+    )
+      .map(({ event }) => `       - ${event.section.toString()}.${event.method.toString()}\n`)
+      .join("")}`
+  ).to.not.be.undefined;
+  return event!.event as any;
+}
+
+export function expectSubstrateEvents<
+  ApiType extends ApiTypes,
+  Call extends
+    | SubmittableExtrinsic<ApiType>
+    | Promise<SubmittableExtrinsic<ApiType>>
+    | string
+    | Promise<string>,
+  Calls extends Call | Call[],
+  Event extends AugmentedEvents<ApiType>,
+  Section extends keyof Event,
+  Method extends keyof Event[Section],
+  Tuple extends ExtractTuple<Event[Section][Method]>
+>(
+  block: BlockCreationResponse<ApiType, Calls extends Call[] ? Awaited<Call>[] : Awaited<Call>>,
+  section: Section,
+  method: Method,
+  count = 0 // if 0, doesn't check
+): IEvent<Tuple>[] {
+  let events: EventRecord[] = [];
+  if (Array.isArray(block.result)) {
+    block.result.forEach((r, idx) => {
+      const foundEvents = r.events.filter(
+        ({ event }) => event.section.toString() == section && event.method.toString() == method
+      );
+      if (foundEvents.length > 0) {
+        events.push(...foundEvents);
+      }
+    });
+  } else {
+    const foundEvents = block.result.events.filter(
+      ({ event }) => event.section.toString() == section && event.method.toString() == method
+    );
+    if (foundEvents.length > 0) {
+      events.push(...foundEvents);
+    }
+  }
+  expect(events.length > 0).to.not.be.null;
```

Comment:
I guess this file is copied from somewhere else, but this line seems useless, `events.length > 0` can only be true of false, and both of the are `!== null`. See https://www.chaijs.com/api/bdd/#method_null